tests/e2e/event_handler/test_header_serializer.py

```diff
@@ -36,106 +39,179 @@ def lambda_function_url_endpoint(infrastructure: dict) -> str:
+        )
+    )
 
     # THEN
-    assert response.status_code == 200
-    assert response.content == b"Hello world"
-    assert response.headers["content-type"] == "text/plain"
+    assert response.status_code == status_code
+    # response.content is a binary string, needs to be decoded to compare with the real string
+    assert response.content.decode("ascii") == body
 
-    # Only the last header for key "Foo" should be set
-    assert response.headers["Foo"] == "zbr"
+    # Only the last header should be set
+    for key, value in headers.items():
+        assert key in response.headers
+        value = value if isinstance(value, str) else sorted(value)[-1]
+        assert response.headers[key] == value
 
     # Only the last cookie should be set
-    assert "MonsterCookie" in response.cookies.keys()
-    assert "CookieMonster" not in response.cookies.keys()
+    assert len(response.cookies.items()) == 1
+    assert last_cookie.name in response.cookies
+    assert response.cookies.get(last_cookie.name) == last_cookie.value
 
 
 def test_alb_multi_value_headers_serializer(alb_multi_value_header_listener_endpoint):
     # GIVEN
     url = f"{alb_multi_value_header_listener_endpoint}/todos"
+    body = "Hello World"
+    status_code = 200
+    headers = {"Content-Type": "text/plain", "Vary": ["Accept-Encoding", "User-Agent"]}
+    cookies = [
+        Cookie(name="session_id", value=str(uuid4()), secure=True, http_only=True),
+        Cookie(name="ab_experiment", value="3"),
+    ]
 
     # WHEN
-    response = data_fetcher.get_http_response(Request(method="GET", url=url))
+    response = data_fetcher.get_http_response(
+        Request(
+            method="POST",
+            url=url,
+            json={"body": body, "status_code": status_code, "headers": headers, "cookies": list(map(str, cookies))},
+        )
+    )
 
     # THEN
-    assert response.status_code == 200
-    assert response.content == b"Hello world"
-    assert response.headers["content-type"] == "text/plain"
+    assert response.status_code == status_code
+    # response.content is a binary string, needs to be decoded to compare with the real string
+    assert response.content.decode("ascii") == body
 
-    # Only the last header for key "Foo" should be set
-    assert "Foo" in response.headers
-    foo_headers = [x.strip() for x in response.headers["Foo"].split(",")]
-    assert sorted(foo_headers) == ["bar", "zbr"]
+    for key, value in headers.items():
+        assert key in response.headers
+        value = value if isinstance(value, str) else ", ".join(sorted(value))
 
-    # Only the last cookie should be set
-    assert "MonsterCookie" in response.cookies.keys()
-    assert "CookieMonster" in response.cookies.keys()
+        # ALB sorts the header values randomly, so we have to re-order them for comparison here
+        returned_value = ", ".join(sorted(response.headers[key].split(", ")))
```

Comment:
Note for us: maybe we should have an utility function for this as we'll forget at some point (syntax et al)